docs/Configuration.md

```diff
@@ -95,6 +95,24 @@ A complete breakdown can be found in the IAM User Guide's
 It is **highly recommended** that you use SSL for encryption when using `PLAIN` or `AWS`,
 otherwise credentials will be transmitted in cleartext!
 
+### SASL Provider Functions
+
+Alternatively, you may set the `sasl` property to a function that returns a compatible configuration
+object, which allows dynamic refresh of credentials at runtime.
+
+```javascript
+function getCredentials() {
```

Comment:
This would have to become something like:

```
async function getCredentials() {
  const credentials = await getToken()
  return credentials
}
```